### PR TITLE
Be louder when graphics is missing for geospatial

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

## Summary

Users building from source could get in a situation where dependencies of the `graphics` component weren't available, so that the component is not created.  In this case, the `geospatial` component still has a hard dependency on the `graphics` component, but it is still attempting to build.

This adds gating logic to prevent the `geospatial` component from building when the `graphics` component isn't build and adds a warning message for the user.

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by Ryanf55 — APPROVED** [review]
The warning is much more obvious and it skips building geospatial if graphics do not exist. Thanks!

image

I'll create another bug for the follow up (assuming it's optional) since it's a different problem.

File: geospatial/src/CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
+if (TARGET ${PROJECT_LIBRARY_TARGET_NAME}-graphics)
+  gz_get_libsources_and_unittests(sources gtest_sources)
+  gz_add_component(geospatial
+    SOURCES ${sources}
+    DEPENDS_ON_COMPONENTS graphics
+    GET_TARGET_NAME geospatial_target)
 
-gz_add_component(geospatial
-  SOURCES ${sources}
-  DEPENDS_ON_COMPONENTS graphics
```
Comment (scpeters) [inline geospatial/src/CMakeLists.txt:5]: I thought the `DEPENDS_ON_COMPONENTS` parameter would generate a complaint if the needed component wasn't available, but apparently it doesn't

**Comment by scpeters** [comment]
an alternative would be to change every `find_package` call with `REQUIRED_BY graphics` to `REQUIRED_BY graphics geospatial`; not sure that it makes more sense though

**Review by scpeters — APPROVED** [review]

File: geospatial/src/CMakeLists.txt (hunk 1 above)
Comment (mjcarroll) [reply geospatial/src/CMakeLists.txt:5]: That was also my understanding, but does not seem to be the case.

**Comment by mjcarroll** [comment]
> an alternative would be to change every find_package call with REQUIRED_BY graphics to REQUIRED_BY graphics geospatial; not sure that it makes more sense though

It would be the other way around, correct?  Anyone who requires geospatial also will require graphics.

**Review by bperseghetti — APPROVED** [review]

**Pull request merged**